### ISSUE TITLE
Error in `MeshStorage::mesh_free` when trying to delete MeshInstance3D with Skeleton3D

### ISSUE THREAD

**Issue by bearman92** [godotengine/godot#62807]
### Godot version

v4.0.alpha.custom_build [09e12ba9b]

### System information

Windows 10 x64

### Issue description

Editor crashes when trying to delete node, that contains mesh with skeleton in editor or runtime.

```
================================================================
CrashHandlerException: Program crashed
Engine version: Godot Engine v4.0.alpha.custom_build (09e12ba9b46934c32ffc9e178b5c6f0348de809f)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[0] Dependency::deleted_notify (D:\git\godot_upstream\servers\rendering\storage\utilities.cpp:42)
[1] RendererRD::MeshStorage::mesh_free (D:\git\godot_upstream\servers\rendering\renderer_rd\storage_rd\mesh_storage.cpp:214)
[2] RendererSceneCull::_instance_update_mesh_instance (D:\git\godot_upstream\servers\rendering\renderer_scene_cull.cpp:474)
[3] RendererSceneCull::instance_attach_skeleton (D:\git\godot_upstream\servers\rendering\renderer_scene_cull.cpp:1012)
[4] RenderingServerDefault::instance_attach_skeleton (D:\git\godot_upstream\servers\rendering\rendering_server_default.h:751)
[5] VisualInstance3D::_notification (D:\git\godot_upstream\scene\3d\visual_instance_3d.cpp:71)
[6] GeometryInstance3D::_notificationv (D:\git\godot_upstream\scene\3d\visual_instance_3d.h:79)
[7] MeshInstance3D::_notificationv (D:\git\godot_upstream\scene\3d\mesh_instance_3d.h:40)
[8] Object::notification (D:\git\godot_upstream\core\object\object.cpp:743)
[9] Node3D::_notification (D:\git\godot_upstream\scene\3d\node_3d.cpp:162)
[10] GeometryInstance3D::_notificationv (D:\git\godot_upstream\scene\3d\visual_instance_3d.h:79)
[11] MeshInstance3D::_notificationv (D:\git\godot_upstream\scene\3d\mesh_instance_3d.h:40)
[12] Object::notification (D:\git\godot_upstream\core\object\object.cpp:743)
[13] Node::_propagate_exit_tree (D:\git\godot_upstream\scene\main\node.cpp:296)
[14] Node::_propagate_exit_tree (D:\git\godot_upstream\scene\main\node.cpp:285)
[15] Node::_propagate_exit_tree (D:\git\godot_upstream\scene\main\node.cpp:285)
[16] Node::_propagate_exit_tree (D:\git\godot_upstream\scene\main\node.cpp:285)
[17] Node::remove_child (D:\git\godot_upstream\scene\main\node.cpp:1215)
[18] call_with_variant_args_dv<EditorNode,Node *> (D:\git\godot_upstream\core\variant\binder_common.h:406)
[19] MethodBindT<EditorNode,Node *>::call (D:\git\godot_upstream\core\object\method_bind.h:318)
[20] Object::callp (D:\git\godot_upstream\core\object\object.cpp:684)
[21] UndoRedo::_process_operation_list (D:\git\godot_upstream\core\object\undo_redo.cpp:324)
[22] UndoRedo::_redo (D:\git\godot_upstream\core\object\undo_redo.cpp:77)
[23] UndoRedo::commit_action (D:\git\godot_upstream\core\object\undo_redo.cpp:297)
[24] SceneTreeDock::_delete_confirm (D:\git\godot_upstream\editor\scene_tree_dock.cpp:2110)
[25] CallableCustomMethodPointer<SceneTreeDock,bool>::call (D:\git\godot_upstream\core\object\callable_method_pointer.h:105)
[26] Callable::call (D:\git\godot_upstream\core\variant\callable.cpp:51)
[27] Object::emit_signalp (D:\git\godot_upstream\core\object\object.cpp:1014)
[28] AcceptDialog::_ok_pressed (D:\git\godot_upstream\scene\gui\dialogs.cpp:107)
[29] Callable::call (D:\git\godot_upstream\core\variant\callable.cpp:51)
[30] Object::emit_signalp (D:\git\godot_upstream\core\object\object.cpp:1014)
[31] BaseButton::_pressed (D:\git\godot_upstream\scene\gui\base_button.cpp:134)
[32] BaseButton::on_action_event (D:\git\godot_upstream\scene\gui\base_button.cpp:176)
[33] BaseButton::gui_input (D:\git\godot_upstream\scene\gui\base_button.cpp:67)
[34] Control::_call_gui_input (D:\git\godot_upstream\scene\gui\control.cpp:953)
[35] Viewport::_gui_call_input (D:\git\godot_upstream\scene\main\viewport.cpp:1281)
[36] Viewport::_gui_input_event (D:\git\godot_upstream\scene\main\viewport.cpp:1589)
[37] Viewport::push_input (D:\git\godot_upstream\scene\main\viewport.cpp:2721)
[38] Window::_window_input (D:\git\godot_upstream\scene\main\window.cpp:1021)
[39] CallableCustomMethodPointer<Window,Ref<InputEvent> const &>::call (D:\git\godot_upstream\core\object\callable_method_pointer.h:104)
[40] Callable::call (D:\git\godot_upstream\core\variant\callable.cpp:51)
[41] DisplayServerWindows::_dispatch_input_event (D:\git\godot_upstream\platform\windows\display_server_windows.cpp:2079)
[42] Input::_parse_input_event_impl (D:\git\godot_upstream\core\input\input.cpp:663)
[43] Input::flush_buffered_events (D:\git\godot_upstream\core\input\input.cpp:889)
[44] DisplayServerWindows::process_events (D:\git\godot_upstream\platform\windows\display_server_windows.cpp:1786)
[45] OS_Windows::run (D:\git\godot_upstream\platform\windows\os_windows.cpp:777)
[46] widechar_main (D:\git\godot_upstream\platform\windows\godot_windows.cpp:179)
[47] _main (D:\git\godot_upstream\platform\windows\godot_windows.cpp:203)
[48] main (D:\git\godot_upstream\platform\windows\godot_windows.cpp:215)
[49] __scrt_common_main_seh (d:\a01\_work\20\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[50] BaseThreadInitThunk
-- END OF BACKTRACE --
================================================================
```

### Steps to reproduce

Run the minimal reproduction project and try to delete Suzanne node.

### Minimal reproduction project

[MeshInstanceCrash.zip](https://github.com/godotengine/godot/files/9060706/MeshInstanceCrash.zip)

**Comment by akien-mga** [comment]
This was fixed by #62784. Now it should no longer crash but raise an error that the mesh is null.

Let's keep this issue open to debug that error further as it seems to be a recent regression (maybe from #62344? @BastiaanOlij).

**Comment by BastiaanOlij** [comment]
This should be fixed by #62947 , really dumb mistake by me :)
